### PR TITLE
front: fix webrtc webcam feed on Firefox

### LINKED ISSUE #555: Firefox does not show webcam permission dialog
When showing the webcam selection site, my firefox 84.0.1 (64-Bit) does not show the dialog for the webcam permissions. It's only asking for microphone permissions. I figured out, that the problem is in the file front/src/WebRtc/MediaManager.ts:
```
let videoConstraint: boolean|MediaTrackConstraints = {
    width: { min: 640, ideal: 1280, max: 1920 },
    height: { min: 400, ideal: 720 },
    frameRate: {exact: valueVideo, ideal: valueVideo},
    facingMode: "user",
    resizeMode: 'crop-and-scale',
    aspectRatio: 1.777777778
};
```
If I remove the exact part from the frameRate constraint, everything works without any problems. This only occours with my firefox, chrome works without any problems.

### PR DESCRIPTION
**Pull request by Informatic**
Fixes #555

### CODE REVIEW

**Comment by gparant** [comment]
I test your MR and I stay bloqued 😭

**Comment by Informatic** [comment]
Could you please rephrase your comment? Does this **not** fix camera stream in Firefox, or does this break other browsers?

**Comment by Informatic** [comment]
FYI i've just verified and it seems to work for me on Firefox 83 and Chromium 87, both on Linux (NixOS 20.09). Camera preview seems to lock up after a while on `EnableCameraScene`, but that is out of scope of this specific issue, I think...

**Comment by tbnobody** [comment]
I also tested this with Firefox/Chrome and Edge on Windows (10) --> works. I mean the camera preview is still black sometimes but it's asking for camera permissions, which it doesn't without this fix.

**Comment by lunar-debian** [comment]
I confirm the fix to work as well. My understanding is that by specifying `exact`, Firefox will consider it has no video stream to offer if no camera is available supporting the given framerate. Chrome seems to have an extra layer when handling videos making it more accommodating.

File: front/src/WebRtc/MediaManager.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ if(localValueVideo){
 let videoConstraint: boolean|MediaTrackConstraints = {
     width: { min: 640, ideal: 1280, max: 1920 },
     height: { min: 400, ideal: 720 },
-    frameRate: {exact: valueVideo, ideal: valueVideo},
+    frameRate: { ideal: valueVideo },
```
Comment (Palid) [inline front/src/WebRtc/MediaManager.ts:16]: In RC3 we had just ```frameRate: valueVideo,```, not as an object. It worked, but seemed to have similar issues like people are reporting in comments.

Comment (Informatic) [reply front/src/WebRtc/MediaManager.ts:16]: https://developer.mozilla.org/en-US/docs/Web/API/ConstrainULong
> In addition, you can specify the value as a simple long integer value, in which case the user agent does its best to match the value once all other more stringent constraints are met.

Not sure if interpreting this correctly, but this seems like passing a plain `Number` here is equivalent to passing an object with just the `ideal` specified. I chose the latter to match other constraints specified already.

**Comment by P1tt187** [comment]
I tryed the fix on my local machine. First it seems to work, but when reloading the page the system picks the wrong camera if mutliple are present. 
This can be the case if you're using virtual devices, e.g. OBS.

**Comment by Informatic** [comment]
This might be related to https://github.com/thecodingmachine/workadventure/pull/581#issuecomment-759047306 and that requires bigger cleanup in media handling.

**Comment by moufmouf** [comment]
I can confirm this works too.
I'll merge this. Thanks a lot @Informatic!

@P1tt187 if you are still seeing a problem with the system picking the wrong camera, could you please open an issue?

**Comment by cweiske** [comment]
When will this be on the live server (play.workadventu.re)?

**Comment by cweiske** [comment]
It's live at least since yesterday.
